packages/image_picker/image_picker_platform_interface/lib/src/types/media_options.dart

```diff
@@ -13,11 +13,15 @@ class MediaOptions {
   const MediaOptions({
     this.imageOptions = const ImageOptions(),
     required this.allowMultiple,
+    this.limit,
```

Comment:
What happens when this object is created with `allowMultiple` as `false`, but `limit` is not `null`?

Reply:
It's firstly verify if the allowMultiple is true and only then query for photos with specified limit

Reply:
So if the `allowMultiple` is `false`, then it allow to select only one file

Reply:
Probably worth throwing a test together to validate this so it doesn't change in the future.

Reply:
Yeah, but I'm also wondering if it's not to much open approach. Maybe be more strict would be less confusing to users? So to require that `allowMultiple` should be `true` when limit is not `null` and also remove `MediaOptions.createAndValidate` and call `_validate` in default constructor?

It will look like that:

```dart
MediaOptions({
  this.imageOptions = const ImageOptions(),
  required this.allowMultiple,
  this.limit,
}) {
  _validate(allowMultiple: allowMultiple, limit: limit);
}

static void _validate({required bool allowMultiple, int? limit}) {
  if (!allowMultiple && limit != null) {
    throw ArgumentError.value(
      allowMultiple,
      'allowMultiple',
      'cannot be false, when limit is not null',
    );
  }

  if (limit != null && limit < 2) {
    throw ArgumentError.value(limit, 'limit', 'cannot be lower then 2');
  }
}
```

What do you think?

Reply:
I'm torn about it. I think throwing when we don't need to (there is a logical option) is not great. Being more strict is also good though.

@stuartmorgan what do you think?

Reply:
Another option is to create a new MediaOptions.withLimit constructor and remove the limit passing from the default constructor:

```dart
MediaOptions({
  this.imageOptions = const ImageOptions(),
  required this.allowMultiple,
}) : limit = null;

MediaOptions.withLimit({
  this.imageOptions = const ImageOptions(),
  required this.limit,
}) : allowMultiple = true; 
```

Reply:
I would throw for invalid combinations of parameters. Specialized constructors aren't a great option for a class that we expect to grow over time, because the combinatorics can become quite bad over time.

Reply:
I've updated PR with a version that throws for invalid parameter combinations.

Reply:
Adding body with validate to the default MediaOptions constructor makes it non const, so it'll break the contract for other packages (web, windows, mac, linux). Finally, I've added to the MediaOptions createAndValidate constructor, the same as it's done for ImageOptions, and mimic it's behaviour in further implementation to be consistent.